pkg/bmc/idrac_virtualmedia.go

```diff
@@ -10,18 +14,29 @@ func init() {
 }
 
 func newRedfishiDracVirtualMediaAccessDetails(parsedURL *url.URL, disableCertificateVerification bool) (AccessDetails, error) {
+	tlsInfo := transport.TLSInfo{
+		TrustedCAFile:      "",
+		InsecureSkipVerify: disableCertificateVerification,
+	}
+	tlsTransport, err := transport.NewTransport(tlsInfo, time.Second*30)
+
+	var client = http.Client{
+		Transport: tlsTransport,
+	}
 	return &redfishiDracVirtualMediaAccessDetails{
```

Comment:
Isn't the above section going to create a new connection for every reconcile loop of each distinct host?

Reply:
done, 
I've add a `bmcClient` to the `ironicProvisioner` struct and created a var `bmcClientSingleton` .
I've added some conditions to not create a new `bmcClient` like we do using `clientIronicSingleton` and `clientInspectorSingleton` using the `bmcClientSingleton`.